src/mongo/db/query/explain_common.cpp

```diff
@@ -51,12 +51,17 @@ const char* ExplainCommon::verbosityString(ExplainCommon::Verbosity verbosity) {
 
 // static
 Status ExplainCommon::parseCmdBSON(const BSONObj& cmdObj, ExplainCommon::Verbosity* verbosity) {
+
     if (Object != cmdObj.firstElement().type()) {
         return Status(ErrorCodes::BadValue, "explain command requires a nested object");
     }
 
     *verbosity = ExplainCommon::EXEC_ALL_PLANS;
-    if (!cmdObj["verbosity"].eoo()) {
+    if (cmdObj.nFields() > 2) {
```

Comment:
Instead of checking how many fields there are, let's loop over all of the fields and make sure that each one is recognized. I think this will be a little clearer. Here's some example code that does a similar thing for an aggregation command (https://docs.mongodb.com/manual/reference/command/aggregate/): https://github.com/mongodb/mongo/blob/r3.5.3/src/mongo/db/pipeline/aggregation_request.cpp#L91-L166

I think you will also have to do something similar and ignore the following field names, which will be parsed elsewhere: "explain", anything that starts with a "$", and that might be it. But I'll check with others here and get back to you.